impeller/renderer/testing/mocks.h

```diff
@@ -164,5 +169,113 @@ class MockTexture : public Texture {
               (override));
 };
 
+class MockTypographerContext : public TypographerContext {
+ public:
+  MOCK_METHOD(std::shared_ptr<GlyphAtlas>,
+              CreateGlyphAtlas,
+              (Context & context,
+               GlyphAtlas::Type type,
+               std::shared_ptr<GlyphAtlasContext> atlas_context,
+               const FontGlyphMap& font_glyph_map),
+              (const, override));
+  MOCK_METHOD(std::shared_ptr<GlyphAtlasContext>,
+              CreateGlyphAtlasContext,
+              (),
+              (const, override));
+};
+
+class MockRenderTargetAllocator : public RenderTargetAllocator {
+ public:
+  MockRenderTargetAllocator(std::shared_ptr<Allocator> allocator)
+      : RenderTargetAllocator(allocator) {}
+  MOCK_METHOD(std::shared_ptr<Texture>,
+              CreateTexture,
+              (const TextureDescriptor& desc),
+              (override));
+};
+
+class MockCapabilities : public Capabilities {
```

Comment:
You don't need to MockCapabilities, there is a CapabilitiesBuilder/StandardCapabilities that lets you set whatever you want.

Reply:
I just removed all this work for now.  They aren't used.  I figured it didn't hurt and may potentially save time in the future.

Reply:
I'll probably end up landing most of this with the other blur anyways where it is actually used.  We'll see.